cristin-project/src/main/java/no/unit/nva/cristin/projects/ProjectPatchValidator.java

```diff
@@ -0,0 +1,141 @@
+        }
+    }
+
+    private static void validateStatus(ObjectNode input) throws BadRequestException {
+        validateNotNullIfPresent(input, no.unit.nva.cristin.model.JsonPropertyNames.STATUS);
+        ProjectStatus.isValidStatus(input.get(no.unit.nva.cristin.model.JsonPropertyNames.STATUS).asText());
+    }
+
+    private static void validateAlternativeTitles(ObjectNode input)
+            throws BadRequestException {
+        validateNotNullIfPresent(input,ALTERNATIVE_TITLES);
+        TypeReference<List<Map<String, String>>> typeRef = new TypeReference<>() {
+        };
+        attempt(() -> OBJECT_MAPPER.readValue(input.get(ALTERNATIVE_TITLES).asText(), typeRef))
+                .orElseThrow(fail ->
+                        new BadRequestException(String.format(ILLEGAL_VALUE_FOR_PROPERTY, ALTERNATIVE_TITLES)));
+    }
+
+    private static void validateLanguageStringMap(ObjectNode input, String propertyName) throws BadRequestException {
+        validateNotNullIfPresent(input, propertyName);
+        TypeReference<Map<String, String>> typeRef = new TypeReference<>() {
+        };
+        try {
+            Map<String, String> languageAndStringMap =
+                    OBJECT_MAPPER.readValue(input.get(propertyName).asText(), typeRef);
+            for (String languageCode : languageAndStringMap.keySet()) {
+                validateLanguageCode(languageCode);
+                validateLanguageContent(languageAndStringMap.get(languageCode), propertyName);
+            }
+        } catch (JsonProcessingException e) {
+            throw new BadRequestException(String.format(ILLEGAL_VALUE_FOR_PROPERTY, propertyName));
+        }
+    }
+
+    private static void validateLanguageContent(String content, String propertyName) throws BadRequestException {
+        if (StringUtils.isEmpty(content)) {
+            throw new BadRequestException(String.format(FIELD_CAN_NOT_BE_ERASED, propertyName));
+        }
+    }
+
+    private static void validateLanguage(ObjectNode input) throws BadRequestException {
+        validateNotNullIfPresent(input, LANGUAGE);
+        validateLanguageUri(input.get(LANGUAGE));
+    }
+
+    private static void validateLanguageCode(String languageCode) throws BadRequestException {
+        if (getLanguageByIso6391Code(languageCode).equals(UNDEFINED_LANGUAGE)) {
+            throw new BadRequestException(String.format(ILLEGAL_VALUE_FOR_PROPERTY, LANGUAGE));
+        }
+    }
+
+    private static void validateLanguageUri(JsonNode jsonNode) throws BadRequestException {
+        if (getLanguageByUri(URI.create(jsonNode.textValue())).equals(UNDEFINED_LANGUAGE)) {
+            throw new BadRequestException(String.format(ILLEGAL_VALUE_FOR_PROPERTY, LANGUAGE));
+        }
+    }
+
+    private static void validateInstant(ObjectNode input, String propertyName) throws BadRequestException {
+        attempt(() -> Instant.parse(input.get(propertyName).asText()))
```

Comment:
What if `propertyName` is not present? Will a missing enddate throw an exception?